internal/controller/postgrescluster/cluster_test.go

```diff
@@ -701,3 +702,59 @@ func TestContainerSecurityContext(t *testing.T) {
 		}
 	}
 }
+
+func TestGenerateClusterReplicaServiceIntent(t *testing.T) {
+	cluster := &v1beta1.PostgresCluster{}
+	cluster.Namespace = "ns1"
+	cluster.Name = "pg2"
+	cluster.Spec.Port = initialize.Int32(9876)
+
+	service := generateClusterReplicaServiceIntent(cluster)
+
+	assert.Assert(t, marshalMatches(service.TypeMeta, `
+apiVersion: v1
+kind: Service
+	`))
+	assert.Assert(t, marshalMatches(service.ObjectMeta, `
+creationTimestamp: null
+labels:
+  postgres-operator.crunchydata.com/cluster: pg2
+  postgres-operator.crunchydata.com/role: replica
+name: pg2-replicas
+namespace: ns1
+	`))
+	assert.Assert(t, marshalMatches(service.Spec, `
```

Comment:
I like the `marshalMatches` here. Definitely want to make use of this for `generate*Intent` functions :100: